CRM/Civicase/Hook/BuildForm/TokenTree.php

```diff
@@ -0,0 +1,392 @@
+<?php
+
+/**
+ * Attaches a new token tree to the form.
+ */
+class CRM_Civicase_Hook_BuildForm_TokenTree {
+
+  const CASE_ROLE_TOKEN_TEXT = 'Case Roles';
+
+  const CURRENT_USER_TOKEN_TEXT = 'Current User';
+
+  const CASE_TOKEN_TEXT = 'Case';
+
+  const CONTACT_TOKEN_TEXT = 'Contact';
+
+  const CLIENT_TOKEN_TEXT = 'Client';
+
+  const OTHER_TOKEN_TEXT = 'Other';
+
+  const ADDRESS_TOKEN_TEXT = 'Address';
+
+  /**
+   * All case and contact related custom fields.
+   *
+   * @var array
+   */
+  private $customFields = [];
+
+  /**
+   * Attaches a new token tree to the form.
+   *
+   * @param CRM_Core_Form $form
+   *   Form object class.
+   * @param string $formName
+   *   Form name.
+   */
+  public function run(CRM_Core_Form &$form, $formName) {
+    if (!$this->shouldRun($formName)) {
+      return;
+    }
+    $this->fetchAllRelevantCustomFields();
```

Comment:
This function more or less seems to set the custom fields to an internal class variable and does not return any value. We should probably rename to `setAllRelevantCustomFields`

Reply:
changed